bindings/rust/s2n-tls/src/error.rs

```diff
@@ -39,7 +40,7 @@ impl From<libc::c_int> for ErrorType {
 #[derive(PartialEq)]
 pub enum Error {
     InvalidInput,
-    Code(s2n_status_code::Type),
+    Code(s2n_status_code::Type, Errno),
```

Comment:
I'm not sure we should have `Errno` as part of our public API. It might be better to do a new type around both `s2n_status_code::Type` and `Errno`, just so we're protected from breaking changes.

Reply:
After some discussion, decided that no details of Error should be public.

So we'll make one breaking change to avoid future breaking changes. Error is now an opaque struct wrapping the original enum, hiding the implementation details. I also gave InvalidInput a kind() (UsageError) so that it can be treated like other errors, and added a source() to more explicitly preserve the original distinction between the two enum variants.